src/storage/archive_index.rs

```diff
@@ -155,18 +186,70 @@ pub(crate) fn find_in_slice(bytes: &[u8], search_for: &str) -> Result<Option<Fil
     .deserialize(&mut deserializer)?)
 }
 
+fn find_in_sqlite_index(conn: &Connection, search_for: &str) -> Result<Option<FileInfo>> {
+    let mut stmt = conn.prepare(
+        "
+        SELECT start, end, compression 
+        FROM files 
+        WHERE path = ?
+        ",
+    )?;
+
+    stmt.query_row((search_for,), |row| {
+        let compression: i32 = row.get(2)?;
+        Ok(FileInfo {
+            range: row.get(0)?..=row.get(1)?,
+            compression: compression.try_into().expect("invalid compression value"),
```

Comment:
This `expect` should be a `context` to just error instead of panicking.

Reply:
Due to the error type in the `TryFrom<i32>` impl for `CompressionAlgorithm`, we can't use `context` here, 

also the return type of the closure `query_row` closure is `rusqlite::Error, which limits the errors that i can use here. 

We _could_ move generating the `FileInfo` outside this closure, but also `FromSqlConversionFailure` semantically matches what we are trying to do here (IMO), to I added a `map_err`